<!-- launch file for Yonah's Iridium Short-Burst-Data (SBD) service (for testing purposes)-->

<launch>

    <!-- screen = output to roslaunch screen for debug; log = log to rosout-->
    <arg name="log_output" default="screen"/>

    <!-- Credentials FIle. Eventually to be moved to despatcher node-->
    <arg name="credentials" default="$(find sbd)/src/login.txt"/>

    <!-- Sleep interval between checks for MO msg -->
    <arg name="interval_3" default="5"/>

    <!-- Serial port that Rockblock is connected to-->
    <arg name="portID" default="/dev/ttyUSB0"/>

    <!-- Rockblock serial no of ourselves and the client-->
    <arg name="own_serial" default="12345"/>
    <arg name="client_serial" default="12345"/>

    <!-- sbd link node -->
    <node pkg="sbd" type="sbd_gnd_link.py" name="sbd_gnd_link"
    respawn="true" respawn_delay="3" output="$(arg log_output)">
        <param name="interval" value="$(arg interval_3)" />
        <param name="credentials" value="$(arg credentials)" />
        <param name="portID" value="$(arg portID)" />
        <param name="own_serial" value="$(arg own_serial)" />
        <param name="client_serial" value="$(arg client_serial)" />
    </node>

    <!-- ground_despatcher node -->
    <node pkg="despatcher" type="gnd_despatcher.py" name="gnd_despatcher"
    respawn="true" respawn_delay="3" output="$(arg log_output)">
    </node>

    <!-- ground test node -->
    <node pkg="despatcher" type="gndtest.py" name="gnd_test"
    respawn="true" respawn_delay="3" output="$(arg log_output)">
    </node>

</launch>
